umh.docs.umh.app/content/en/docs/production-guide/upgrading/0.9.12.md

```diff
@@ -0,0 +1,262 @@
+---
+title: "Upgrade to v0.9.12"
+content_type: upgrading
+description: "This page describes how to upgrade the United Manufacturing Hub
+to version 0.9.12"
+weight: 1000
+---
+
+This page describes how to upgrade the United Manufacturing Hub to version
+0.9.12. Before upgrading, remember to backup the
+[database](/docs/production-guide/backup_recovery/backup-timescale/),
+[Node-RED flows](/docs/production-guide/backup_recovery/import-export-node-red/),
+and your cluster configuration.
+
+## Add Helm repo in {{% resource type="lens" name="name" %}}
+
+Check if the UMH Helm repository is added in {{% resource type="lens" name="name" %}}.
+To do so, from the top-left menu, select **FIle** > **Preferences** (or press `CTRL + ,`).
+Click on the **Kubernetes** tab and check if the **Helm Chart** section contains
+the `https://repo.umh.app` repository.
+
+If it doesn't, click the **Add Custom Helm Repo** button and fill in the following
+values:
+
+- Helm repo name: {{% resource type="helm" name="repo" %}}
+- URL: {{% resource type="helm" name="repo-url" %}}
+
+Then click **Add**.
+
+<!-- Add here any steps needed before deleting the deployments and statefulsets -->
+## Backup RBAC configuration for MQTT Broker
+
+This step is only needed if you enabled RBAC for the MQTT Broker and changed the
+default password. If you did not change the default password, you can skip this
+step.
+
+1. Navigate to **Config** > **ConfigMaps**.
+2. Select the {{% resource type="configmap" name="mqttbroker-credentials" %}}
+   ConfigMap.
+3. Copy the content of `credentials.xml` and save it in a safe place.
+
+## Clear Workloads
+
+Some workloads need to be deleted before upgrading. This process _do not_ delete
+any data. If a workload is missing, it means that it was not enabled in your
+cluster, therefore you can skip it.
+
+To delete a resource, you can select it using the box on the left of the
+resource name and click the **-** button on the bottom right corner.
+
+1. Open the **Workloads** tab.
+2. From the **Deployment** section, delete the following deployments:
+   - {{% resource type="deployment" name="barcodereader" %}}
+   - {{% resource type="deployment" name="factoryinsight" %}}
+   - {{% resource type="deployment" name="kafkatopostgresql" %}}
+   - {{% resource type="deployment" name="mqttkafkabridge" %}}
+   - {{% resource type="deployment" name="mqttsimulator" %}}
+   - {{% resource type="deployment" name="opcuasimulator" %}}
+3. From the **StatefulSet** section, delete the following statefulsets:
+   - {{% resource type="statefulset" name="mqttbridge" %}}
+   - {{% resource type="statefulset" name="mqttbroker" %}}
+   - {{% resource type="statefulset" name="nodered" %}}
+   - {{% resource type="statefulset" name="sensorconnect" %}}
+
+<!-- Add here any steps needed before upgrading the Helm Chart -->
+
+## Remove MQTT Broker extension PVC
+
+In this version we reduced the size of the MQTT Broker extension PVC. To do so,
```

Comment:
if you are using k3s, you can skip this step. by default, k3s with local patgs does not enforce the size of a local-path volume

Reply:
Source for this? I can't find anything about it in the k3s docs, and testing with k3d results in an error when trying to use a pvc with a lower size

Reply:
https://github.com/rancher/local-path-provisioner

> No support for the volume capacity limit currently.
> The capacity limit will be ignored for now.